### PR TITLE
(test) TRUNK-6603: Add unit tests for ProviderService APIException

### PR DESCRIPTION
**Pull request by AshThe25**
## Description
This code adds unit tests for getProviderByUuid. The tests ensure that the method behaves correctly if a provider is found or not.

## Tests
- Returns provider if found
- Throws APIException if provider is null

## Context
This code is an extension of the previous fix that prevents silent returns of null.

### CODE REVIEW

File: api/src/main/java/org/openmrs/api/impl/ProviderServiceImpl.java [hunk 1]
```diff
@@ -119,10 +119,16 @@ public Provider saveProvider(Provider provider) {
...
-		return dao.getProviderByUuid(uuid);
-	}
+@Transactional(readOnly = true)
+public Provider getProviderByUuid(String uuid) {
+    Provider provider = dao.getProviderByUuid(uuid);
+
+    if (provider == null) {
+        throw new APIException("Provider not found for uuid: " + uuid);
+    }
+
```
Comment (tendomart) [inline api/src/main/java/org/openmrs/api/impl/ProviderServiceImpl.java]: Remove extra spaces before and after this if() statement

File: api/src/test/java/org/openmrs/api/impl/ProviderServiceImplTest.java [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.*;
```
Comment (tendomart) [inline api/src/test/java/org/openmrs/api/impl/ProviderServiceImplTest.java]: Remove wild cards, it's a must to do actual imports not generic wild card imports

File: api/src/test/java/org/openmrs/api/impl/ProviderServiceImplTest.java [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+    }
+
+    @Test
+    public void getProviderByUuid_shouldReturnProvider_whenProviderExists() {
+        String uuid = "test-uuid";
+        Provider provider = new Provider();
+
+        when(dao.getProviderByUuid(uuid)).thenReturn(provider);
+
+        Provider result = service.getProviderByUuid(uuid);
```
Comment (tendomart) [inline api/src/test/java/org/openmrs/api/impl/ProviderServiceImplTest.java]: remove extra spaces

File: api/src/test/java/org/openmrs/api/impl/ProviderServiceImplTest.java [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+        assertNotNull(result);
+        assertEquals(provider, result);
+    }
+
+    @Test
+    public void getProviderByUuid_shouldThrowException_whenProviderNotFound() {
+        String uuid = "invalid-uuid";
+
```
Comment (tendomart) [inline api/src/test/java/org/openmrs/api/impl/ProviderServiceImplTest.java:54]: same here

**Comment by tendomart** [comment]
Improve your title by adding The name of the Ticket you're working on, if any, add link to Ticket if any.

**Comment by AshThe25** [comment]
JIRA: https://issues.openmrs.org/browse/TRUNK-6618

No unit tests existed for ProviderServiceImpl.getProviderByUuid. 
Added tests covering the case where a provider exists and the case 
where none is found (APIException expected).

File: api/src/main/java/org/openmrs/api/impl/ProviderServiceImpl.java (hunk 1 above)
Comment (AshThe25) [reply api/src/main/java/org/openmrs/api/impl/ProviderServiceImpl.java]: Thanks for the feedback! I've removed wildcard imports and replaced them with explicit imports.

Also fixed formatting issues and removed extra spaces for consistency.

Updated the PR title for better clarity.

File: api/src/test/java/org/openmrs/api/impl/ProviderServiceImplTest.java (hunk 2 above)
Comment (AshThe25) [reply api/src/test/java/org/openmrs/api/impl/ProviderServiceImplTest.java]: Thanks for the feedback! I've removed wildcard imports and replaced them with explicit imports.

Also fixed formatting issues and removed extra spaces for consistency.

**Comment by Kanishka680** [comment]
Excellent addition of unit tests for edge cases. It's important to verify how the APIException handles null messages.

**Comment by AshThe25** [comment]
> Excellent addition of unit tests for edge cases. It's important to verify how the APIException handles null messages.

THANKS A LOT KANISHKA MEANS A LOT

**Review by Kanishka680 — COMMENTED** [review]
Excellent addition of unit tests for edge cases. It's important to verify how the APIException handles null messages.

**Review by Kanishka680 — COMMENTED** [review]
The implementation of the pluggable validator looks very modular. Great use of the UserService interface.

**Comment by shivvani-r** [comment]
Kindly refer to the [Pull Request Tips](https://openmrs.atlassian.net/wiki/spaces/docs/pages/25477199/Pull+Request+Tips) and make the required changes.

**Review by tendomart — COMMENTED** [review]
your title  can be like this

"TRUNK-6603: Add unit tests for ProviderService APIException"


I see no link to the Title you're working on . Always include it in your description

**Comment by AshThe25** [comment]
Hi @tendomart @Kanishka680! I really apologize for not getting back to you guys earlier, I had a little accident and had to recover for about a week. Anyway, thanks to all the reviewers, and here is the latest commit that contains all necessary changes:

MPL license header added according to project conventions
org.openmrs.api.impl; package import statement added (this might have been the cause of the SonarCloud issue with code coverage)
Import statements corrected: non-static before static in order
Self-import removed (we are inside org.openmrs.api.impl)
Generated realistic UUID by using the project's fixture objects

**Comment by AshThe25** [comment]
JIRA: https://issues.openmrs.org/browse/TRUNK-6618

No unit tests existed for ProviderServiceImpl.getProviderByUuid. 
Added tests covering the case where a provider exists and the case 
where none is found (APIException expected).

**Comment by AshThe25** [comment]
> ## Description
> This code adds unit tests for getProviderByUuid. The tests ensure that the method behaves correctly if a provider is found or not.
> 
> ## Tests
> * Returns provider if found
> * Throws APIException if provider is null
> 
> ## Context
> This code is an extension of the previous fix that prevents silent returns of null.

JIRA: https://issues.openmrs.org/browse/TRUNK-6618

No unit tests existed for ProviderServiceImpl.getProviderByUuid. 
Added tests covering the case where a provider exists and the case 
where none is found (APIException expected).
